lib/compiler/src/beam_asm.erl

```diff
@@ -199,10 +199,16 @@ build_file(Code, Attr, Dict, NumLabels, NumFuncs, ExtraChunks, CompileInfo, Comp
                       <<?BEAM_TYPES_VERSION:32, NumTypes:32>>,
                       TypeTab),
 
+    %% Create the meta chunk
+    Meta = proplists:get_value(<<"Meta">>, ExtraChunks, term_to_binary([])),
+    MetaChunk = chunk(<<"Meta">>, Meta),
+    %% Remove Meta chunk from ExtraChunks since it is essential
+    ExtraChunks1 = ExtraChunks -- [{<<"Meta">>, Meta}],
```

Comment:
We use @rvirding's convention for numbering updated variables in the `compiler` application. Rename `ExtraChunks` in the function header for `build_files` to `ExtraChunks0`. Then:

```suggestion
    Meta = proplists:get_value(<<"Meta">>, ExtraChunks0, term_to_binary([])),
    MetaChunk = chunk(<<"Meta">>, Meta),
    %% Remove Meta chunk from ExtraChunks since it is essential
    ExtraChunks = ExtraChunks0 -- [{<<"Meta">>, Meta}],
```